src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/KNNPlugin.java

```diff
@@ -118,7 +118,8 @@
  */
 public class KNNPlugin extends Plugin implements MapperPlugin, SearchPlugin, ActionPlugin, EnginePlugin, ScriptPlugin {
 
-    public static final String KNN_BASE_URI = "/_opendistro/_knn";
+    public static final String LEGACY_KNN_BASE_URI = "/_opendistro/_knn";
```

Comment:
You probably want this to be `LEGACY_OPENDISTRO_`, because there will be other legacies.